index.js

```diff
@@ -104,15 +104,27 @@ export const buyProduct = (sku) => Platform.select({
   android: () => RNIapModule.buyItemByType(ANDROID_ITEM_TYPE_IAP, sku, null)
 })();
 
+/**
+ * Buy a product with a specified quantity (iOS only)
+ *   Call finishTransaction after receipt validation process.
+ * @param {string} sku The product's sku/ID
+ * @param {number} quantity The amount of product to buy
+ * @returns {Promise<ProductPurchase>}
+ */
+export const buyProductWithQuantity = (sku, quantity) => Platform.select({
+  ios: () => RNIapIos.buyProductWithQuantity(sku, quantity),
+  android: () => RNIapModule.buyItemByType(ANDROID_ITEM_TYPE_IAP, sku, null)
```

Comment:
I think it would be better to just resolve in `android`.
```
() => Promise.resolve(),
```

Just make above method only `ios` dependent. Naming it as `buyProductWithQuantityIOS` will be better too.

Reply:
I agree with `buyProductWithQuantityIOS`, I will update my PR

Reply:
Actually, if the method `buyProductWithoutFinishTransaction` is already iOS only, we should either rename both method or leave `buyProductWithQuantity` what do you think?

Reply:
@lefoy Good point. That might have confused you. `buyProductWithoutFinishTransaction` however works seamlessly with android. The purpose of this method is to avoid omission in response since there can be delays in callback from apple itself waiting for the transaction to complete. Using this feature may be risky in some situation, but some devs wanted it for whatever reason. 
Anyway, I want to confirm you that `buyProductWithoutFinishTransaction` share almost the same feature in each platform with some differences at the back-side. Therefore, it is somewhat different from `buyProductWithQuantity` feature because it actually is serving different parameter that won't be used in `android`.
Therefore, I think it'd be better to name it as `buyProductWithQuantityIOS` and not provide it for android.